app/models/order.rb

```diff
@@ -4,7 +4,26 @@ class Order <ApplicationRecord
   has_many :item_orders
   has_many :items, through: :item_orders
 
+  enum status: %w(default pending)
+
+  def self.find_order(order_id)
+    Order.find_by(id: order_id)
+  end
+
   def grandtotal
     item_orders.sum('price * quantity')
   end
+
+  def quantity_sum_per_merchant(merchant)
+    items.joins(:item_orders).where('items.merchant_id =?', merchant.id).sum('item_orders.quantity')
+  end
+
+  def value_sum_per_merchant(merchant)
+    items.joins(:item_orders)
+         .where('items.merchant_id =?', merchant.id)
+         .group('item_orders.quantity')
+         .sum('item_orders.price')
+         .inject(0) {|result, (key, value)| result += (key * value)}
```

Comment:
This took me approximately 48 hours and now I understand AR waaaaayyyyy better. Happy to share the knowledge. Also, hello `inject` it is so nice to finally meet you!

Reply:
Would love a walk through of this, glad AR is finally clicking!